provider/dagprovider_test.go

```diff
@@ -0,0 +1,88 @@
+package provider
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ipfs/boxo/blockservice"
+	"github.com/ipfs/boxo/blockstore"
+	"github.com/ipfs/boxo/exchange/offline"
+	bsfetcher "github.com/ipfs/boxo/fetcher/impl/blockservice"
+	"github.com/ipfs/boxo/ipld/merkledag"
+	mdutils "github.com/ipfs/boxo/ipld/merkledag/test"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	"github.com/ipfs/go-unixfsnode"
+	dagpb "github.com/ipld/go-codec-dagpb"
+	"github.com/stretchr/testify/require"
+)
+
+func makeDAGProvider(t *testing.T, ctx context.Context) (KeyChanFunc, []cid.Cid) {
+	t.Helper()
+
+	ds := dssync.MutexWrap(datastore.NewMapDatastore())
+	bs := blockstore.NewBlockstore(ds)
+	bserv := blockservice.New(bs, offline.Exchange(bs))
+	ipldFetcher := bsfetcher.NewFetcherConfig(bserv)
+	ipldFetcher.PrototypeChooser = dagpb.AddSupportToChooser(bsfetcher.DefaultPrototypeChooser)
+	unixFSFetcher := ipldFetcher.WithReifier(unixfsnode.Reify)
+	dserv := merkledag.NewDAGService(bserv)
```

Comment:
I have copied from Kubo initialization and I'm not sure what a PrototypeChooser is but whether it matters or not is probably irrelevant here.